crates/bevy_ecs/src/system/mod.rs

```diff
@@ -1,3 +1,68 @@
+//! Tools for controlling behavior in an ECS application.
+//!
+//! Systems define how an ECS based application behave. They have to be registered to a
+//! [`SystemStage`](crate::schedule::SystemStage) to be able to run. A system is usually
+//! written as a normal function that will be automatically converted into a system.
+//!
+//! System functions can have parameters, through which one can query and mutate Bevy ECS state.
+//! Only types that implement [`SystemParam`] can be used, automatically fetching data from
+//! the [`World`](crate::world::World).
+//!
+//! System functions often look like this:
+//!
+//! ```
+//! # use bevy_ecs::prelude::*;
+//! #
+//! # struct Player { alive: bool }
+//! # struct Score(u32);
+//! # struct Round(u32);
+//! #
+//! fn update_score_system(
+//!     mut query: Query<(&Player, &mut Score)>,
+//!     mut round: ResMut<Round>,
+//! ) {
+//!     for (player, mut score) in query.iter_mut() {
+//!         if player.alive {
+//!             score.0 += round.0;
+//!         }
+//!     }
+//!     round.0 += 1;
+//! }
+//! # update_score_system.system();
+//! ```
+//!
+//! # System ordering
+//!
+//! While the execution of systems is usually parallel and not deterministic, there are two
+//! ways to determine a certain degree of execution order:
+//!
+//! - **System Stages:** They determine hard execution synchronization boundaries inside of
+//!   which systems run in parallel by default.
+//! - **Labeling:** First, systems are labeled upon creation by calling `.label()`. Then,
+//!   methods such as `.before()` and `.after()` are appended to systems to determine
+//!   execution order in respect to other systems.
+//!
+//! # System parameter list
+//! Following is the complete list of accepted types as system parameters:
+//!
+//! - All tuples between 1 to 16 elements where each element implements [`SystemParam`]
```

Comment:
Imo these lines should be sorted by "approximate" relevance to help guide users to the "most useful types". The tuples and unit type description should come last. We should start with Query, Res, ResMut, and Commands. I dont care much about the other orderings.